custom_components/waste_collection_schedule/waste_collection_schedule/source/seattle_gov.py

```diff
@@ -21,40 +21,132 @@ def __init__(self, street_address):
+            }
+        }
 
-        # create entries for trash, recycling, and yard waste
+        r = requests.post(
+            f"https://myutilities.seattle.gov/rest/serviceorder/findAccount",
+            json=find_account_payload)
+
+
+        account_info = json.loads(r.text)
+        account_number = account_info["account"]["accountNumber"]
+
+
+
+        #step 3
+        token_payload = {"grant_type": "password", "username":"guest","password":"guest"}
+
+        r = requests.post(
+            f"https://myutilities.seattle.gov/rest/auth/token",
+            data=token_payload)
+       
+        token_info = json.loads(r.text)
+        token = token_info["access_token"]
+
+        headers = {
+            "Authorization": "Bearer {0}".format(token)
+        }
+
+        #step 4
+        swsummary_payload = {
+            "customerId": "guest",
+            "accountContext": {
+                "accountNumber": account_number,
+                "personId": None,
+                "companyCd": None,
+                "serviceAddress": None
+            }
+        }
+
+        r = requests.post(
+            f"https://myutilities.seattle.gov/rest/account/swsummary",
+            json=swsummary_payload,
+            headers=headers)
+
+
+    
+        summary_info = json.loads(r.text)
+        # the description property in each service in swServices it's either 'Garbage', 'Recycle', or 'Food/Yard Waste'
+        swServices = summary_info["accountSummaryType"]["swServices"][0]["services"]
+        personId = summary_info["accountContext"]["personId"]
+        companyCd = summary_info["accountContext"]["companyCd"]
+
+
+        #step 5
+        waste_calendar_payload = {
+            "customerId": "guest",
+            "accountContext": {
+                "accountNumber": account_number,
+                "personId": personId,
+                "companyCd": companyCd
+            },
+            "servicePoints": []
+        }
+
+        #fill out payload
+        for service in swServices:
+            waste_calendar_payload["servicePoints"].append(service["servicePointId"])
+
+
+        r = requests.post(
+            f"https://myutilities.seattle.gov/rest/solidwastecalendar",
+            json=waste_calendar_payload,
+            headers=headers)
+
+
+        calendar_info = {}
         entries = []
 
-        if next_pickup["Garbage"]:
-            entries.append(
-                Collection(date=next_pickup_date, t="Trash", icon="mdi:trash-can")
-            )
-        if next_pickup["FoodAndYardWaste"]:
-            entries.append(
-                Collection(
-                    date=next_pickup_date, t="Food and Yard Waste", icon="mdi:leaf"
+
+        calendar_info = json.loads(r.text)
+
+        for service in swServices:
+            name = service["description"]
+            servicePointId = service["servicePointId"]
+            next_date = datetime.datetime.strptime(calendar_info["calendar"][servicePointId+"_NP"][0], "%m/%d/%Y").date()
```

Comment:
I have seen that the json response also contains a list of all collection dates, not only the upcoming one - in the attribute "servicePointId" (without "_NP"). Please add all available dates because then also the calendar and this list of upcoming collections will be filled nicely.

Reply:
Oh I see. I was following the previous implementation that only included the next one. Is the idea that there would be one collection entry per date rather than just the next one? If yes, I agree that’s better and can make that change

Reply:
fixed in the latest commit

Reply:
Yes, the idea is to have one collection entry per date. The framework will take care which one's will be displayed.